tasks/git/src/main/java/com/walmartlabs/concord/plugins/git/GitHubTaskAction.java

```diff
@@ -0,0 +1,47 @@
+package com.walmartlabs.concord.plugins.git;
+
+import com.walmartlabs.concord.plugins.git.client.GitHubClient;
+import com.walmartlabs.concord.plugins.git.model.GitHubApiInfo;
+
+import java.util.Map;
+import java.util.UUID;
+
+public abstract class GitHubTaskAction<T extends GitHubTaskParams> {
+
+    public abstract Action action();
+
+    public abstract Map<String, Object> execute(UUID txId, GitHubApiInfo apiInfo, T input) throws Exception;
+
+    protected GitHubClient createClient(UUID txId, GitHubApiInfo apiInfo) {
+        return new GitHubClient(txId, apiInfo);
+    }
+
+    public enum Action {
```

Comment:
Do we actually need this enum here? The mapping between a GitHubTaskAction implementor and the `action` variable is done in the task anyway. Maybe the enum should be a private member of the task's class instead?